Description: support terminals bigger than 129 lines
 mg segfaults when the terminal is very wide.
 Using a larger datatype is reported to fix this.
Origin: http://bugs.debian.org/cgi-bin/bugreport.cgi?msg=12;bug=573447
Bug-Debian: http://bugs.debian.org/573447
Forwarded: mailto:han@mijncomputer.nl (Han Boetes)
Reviewed-By: Trent W. Buck <trentbuck@gmail.com>
Last-Update: 2010-03-19
Author: xavier.hienne@free.fr
Index: mg-20110120/def.h
===================================================================
--- mg-20110120.orig/def.h	2011-03-01 15:04:31.418745025 +1100
+++ mg-20110120/def.h	2011-03-01 15:04:48.431449555 +1100
@@ -204,9 +204,9 @@
 	struct line	*w_markp;	/* Line containing "mark"	*/
 	int		 w_doto;	/* Byte offset for "."		*/
 	int		 w_marko;	/* Byte offset for "mark"	*/
-	char		 w_toprow;	/* Origin 0 top row of window	*/
-	char		 w_ntrows;	/* # of rows of text in window	*/
-	char		 w_frame;	/* #lines to reframe by.	*/
+	int		 w_toprow;	/* Origin 0 top row of window	*/
+	int		 w_ntrows;	/* # of rows of text in window	*/
+	int		 w_frame;	/* #lines to reframe by.	*/
 	char		 w_rflag;	/* Redisplay Flags.		*/
 	char		 w_flag;	/* Flags.			*/
 	struct line	*w_wrapline;
